crates/hir-def/src/item_tree/lower.rs

```diff
@@ -662,7 +662,7 @@ impl<'a> Ctx<'a> {
             let attrs = RawAttrs::new(self.db.upcast(), &param, self.body_ctx.span_map());
             debug_assert!(self.generic_param_attr_buffer.insert(item, attrs).is_none());
         };
-
+        self.body_ctx.take_impl_traits_bounds();
```

Comment:
Issue was basically this line missing. Without it, we start allocating impl trait args for things that are not related to the generic params / where clauses